### ISSUE TITLE
[/collector] Add diagrams from Collector's docs/design.md to Configuration and Deployment pages

### ISSUE THREAD

**Issue by mx-psi** [open-telemetry/opentelemetry.io#3547]
**What needs to be changed?** On open-telemetry/opentelemetry-collector/issues/8889 we discussed removing [`docs/design.md`](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/design.md) in favor of the opentelemetry.io documentation. There are some diagrams on that document that would be good to add to our docs before removing that doc.

**What is the name + path of the page that needs changed?** /collector/deployment and /collector/configuration

**Comment by tiffany76** [comment]
Hi! I’d like to work on this issue. 

Here’s my plan. I'm a first-time contributor, so please let me know if I’m going in the right direction.

On https://github.com/open-telemetry/opentelemetry-collector/issues/8889, @cartermp [said](https://github.com/open-telemetry/opentelemetry-collector/issues/8889#issuecomment-1812775064):

> Specifically, I think this can be a top-level doc under the collector folder, perhaps just called architecture. We can then merge in/remove content either as a part of that PR or later.

And @mhausenblas [said](https://github.com/open-telemetry/opentelemetry-collector/issues/8889#issuecomment-1812784817):

> +1 to docs/collector/architecture and maybe (in terms of left-hand side menu) right before "Building a custom collector"?

Guided by these comments, I will do the following:
- Create a new doc under the [Collector folder](https://github.com/open-telemetry/opentelemetry.io/tree/main/content/en/docs/collector).
- Name it `Architecture`.
- Insert the sidebar navigation to the new page between “Transforming telemetry” and “Building a custom collector.” 
![Screen Shot 2024-02-14 at 6 06 48 PM](https://github.com/open-telemetry/opentelemetry.io/assets/30397949/181ddb7a-639a-423a-954b-685777b33df7)
- Copy the text as-is from [design.md](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/design.md).
- Recreate the diagrams using Mermaid.

Finally, referring back to @cartermp’s comment:

>We can then merge in/remove content either as a part of that PR or later.

I’ll create a new issue for merging the `Architecture` page content with existing docs pages, such as [Configuration](https://opentelemetry.io/docs/collector/configuration/) and [Deployment](https://opentelemetry.io/docs/collector/deployment/), after #3547 is resolved.

If this plan sounds good, may I get started?

**Comment by svrnm** [comment]
Thank you @tiffany76! Your suggested plan sounds good to me, @mx-psi / @cartermp / @mhausenblas  wdyt?

**Comment by mhausenblas** [comment]
> Your suggested plan sounds good to me, @mx-psi / @cartermp / @mhausenblas wdyt?

Perfect, thanks, SGTM and let's do it!

**Comment by cartermp** [comment]
Yep, sounds good to me too!

**Comment by svrnm** [comment]
@tiffany76 please go ahead and create the PR, feel free to share your work early, so we can give you rapid feedback! Thank you!

**Comment by tiffany76** [comment]
> @tiffany76 please go ahead and create the PR, feel free to share your work early, so we can give you rapid feedback! Thank you!

Thanks, @svrnm! My draft PR is up. I'm working on the Mermaid diagrams now.
